Component/LinearSolver/Direct/src/sofa/component/linearsolver/direct/SparseLUSolver.h

```diff
@@ -67,13 +76,19 @@ class SparseLUSolver : public sofa::component::linearsolver::MatrixLinearSolver<
 
     Data<bool> f_verbose; ///< Dump system state at each iteration
     Data<double> f_tol; ///< tolerance of factorization
-
-    SparseLUSolver();
+    
     void solve (Matrix& M, Vector& x, Vector& b) override;
     void invert(Matrix& M) override;
 
+    SparseLUSolver();
+
 protected :
 
+    Data<sofa::helper::OptionsGroup> type_perm;
```

Comment:
All Data must start with `d_`. See https://www.sofa-framework.org/community/doc/programming-with-sofa/guidelines/.
Please rename to `d_typePermutation`.

Reply:
Thanks for adding the `d_` prefix. You also must use lower camel case for variables names (see again the guidelines). In your case, `d_type_perm` must be written `d_typePerm`.  But I find clearer to have full names (permutation instead of perm). So I suggest `d_typePermutation`.